plugins/inline.ts

```diff
@@ -50,7 +50,7 @@ export default function (userOptions?: Options) {
         page.document!.querySelectorAll("template"),
       )
         .flatMap((template) =>
-          Array.from(template.content.querySelectorAll(selector))
+          Array.from(template.querySelectorAll(selector))
```

Comment:
this usage may or may not be unique to linkedom.

Reply:
I think it is - and looks like it's broken, according to https://developer.mozilla.org/en-US/docs/Web/API/HTMLTemplateElement.

I wonder if we ought to do something like the following to deal with this scenario...

```ts
template.content?.querySelectorAll(selector) || template.querySelectorAll(selector)
```

Reply:
Actually, not sure that works...

Reply:
This is a bit more robust than a previous suggestion...
```suggestion
        {
          let inlineElements = Array.from(template.content.querySelectorAll(selector));
          // Test for Linkedom
          if(inlineElements.map(x => x.parentElement).filter(x => x === null).length > 0) {
            inlineElements = Array.from(template.querySelectorAll(selector));
          };
          return inlineElements;
        }
```

Reply:
I don't think the code is correct actually, since `template.content` will move the children nodes to a fragment in linkedom.

One workaround is to do

```
template.childNodes.length === 0 ? template.content.querySelectorAll(_) : template.querySelectorAll(_)
```

I've talked to @WebReflection about this: https://github.com/WebReflection/linkedom/pull/279

Reply:
because `template.content` moves children indeed (once) it's a matter of tradeoffs:

  * are you sure no `.content` has been accessed previously? use `template.qSA`
  * do you want to move `content` ? always reach content
  * do you expect changes to be applied? I still need to understand the use case but this might be on LinkeDOM side of affairs, it's just that I have zero time these days to look at it, I am afraid

Reply:
what is `template.qSA`?

i can't find it in the source code. we use the TypeScript version directly.

Reply:
*qSA* is `template.querySelectorAll`, sorry for the terseness